0013/README.md

```diff
@@ -0,0 +1,181 @@
+---
+name: Data Receiver Protocol
+shortname: 13/DRP
+status: draft
+editor: Andrew Kokhanovskyi <ak@kaaiot.io>
+---
+
+<!-- toc -->
+
+
+# Introduction
+
+Data Receiver Protocol (DRP) is designed to communicate endpoint data samples from transmitter to receiver services in the Kaa platform.
+
+DRP complies with the [Inter-Service Messaging](/0003/README.md) guidelines.
+
+
+# Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+The following terms and definitions are used in this RFC.
+
+- **Endpoint data sample**: block of endpoint-related data of arbitrary structure.
+
+- **Data transmitter service (transmitter)**: service that sends endpoint data samples.
+
+- **Data receiver service (receiver)**: service that receives endpoint data samples for storage, processing, etc.
+
+
+# Design
+
+## Endpoint data samples transmission to receivers
+
+In order to send endpoint data samples to receivers, transmitters MUST [broadcast](/0003/README.md#broadcast-messaging) `EndpointDataSamplesEvent` messages to the following NATS subject:
+```
+kaa.v1.events.<transmitter-service-instance-name>.endpoint.data-collection.data-received
+```
+where `<transmitter-service-instance-name>` is the instance name of the transmitter service.
+
+In case a transmitter expects a processing result message from receivers, it MUST set `replyTo` NATS subject in the broadcast message according to the [session affinity design](/0003/README.md#session-affinity).
+
+The NATS message payload is an Avro object with the following schema ([ep-data-samples-event.avsc](./ep-data-samples-event.avsc)):
+
+```json
+{
+    "namespace":"org.kaaproject.ipc.drp.gen.v1",
+    "type":"record",
+    "name":"EndpointDataSamplesEvent",
+    "doc": "Broadcast event with endpoint data samples to data receivers",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds since the timestamp until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"appVersionName",
+            "type":"string",
+            "doc":"Application version name the data samples are sent for"
+        },
+        {
+            "name":"endpointId",
+            "type":[
+                "string",
+                "null"
```

Comment:
Do we support endpoint-unaware data samples?

Reply:
Not really. I'll make EP ID mandatory.